src/sophys/common/devices/picolo.py

```diff
@@ -9,27 +9,80 @@
     EpicsSignalRO,
 )
 from ophyd.flyers import FlyerInterface
-from ophyd.signal import DEFAULT_WRITE_TIMEOUT
+from ophyd.signal import DEFAULT_WRITE_TIMEOUT, DEFAULT_CONNECTION_TIMEOUT
 from ophyd.status import SubscriptionStatus, StatusBase, Status
 
 from ..utils.status import PremadeStatus
+from ..utils.interfaces import IEnumValidator
 
 
-class PicoloAcquisitionTime(EpicsSignalWithRBV):
-    """Device that handles set enum acquisition time using a float value in ms."""
+class EnumAcquisitionTimeValidator(IEnumValidator):
+    """Validates acquisition time based on predefined enum values."""
 
-    def set(self, acquisition_time: float, **kwargs):
-        enums = self.metadata["enum_strs"]
-        enums_float = [(float(item.replace(" ms", "")) / 1000) for item in enums]
-        if acquisition_time not in enums_float:
+    def __init__(self, enum_string):
+        self._enums = [float(item.replace(" ms", "")) / 1000 for item in enum_string]
+
+    def is_valid(self, enum_value: float) -> bool:
+        return enum_value in self._enums
+
+
+class PicoloAcquisitionTimeBase:
+    """Base class for handling acquisition time."""
+
+    def __init__(self, validator: IEnumValidator):
+        self.validator = validator
+
+    def format_acquisition_time(self, acquisition_time: float) -> str:
+        """Convert float acquisition time to formatted string."""
+        ms_value = acquisition_time * 1000  # Convert seconds to milliseconds
+
+        if ms_value == int(ms_value):
+            return f"{int(ms_value)} ms"
+        else:
+            return f"{ms_value} ms"
+
+    def validate_and_format(self, acquisition_time: float):
+        """Validate and format acquisition time."""
+        if not self.validator.is_valid(acquisition_time):
+            raise ValueError(
+                f"The acquisition time '{acquisition_time}' ms is not a valid option."
+            )
+        return self.format_acquisition_time(acquisition_time)
+
+
+class PicoloAcquisitionTimeBaseMixin:
```

Comment:
This should probably be named `PicoloAcquisitionTimeMixin` instead, as it is not a base anymore, it is the final implementation that'll be added to `AcquisitionTime`s